src/Translations/Visitor/PropertyValidationVisitor.php

```diff
@@ -0,0 +1,96 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\PhpCs\Translations\Visitor;
+
+use Becklyn\PhpCs\Translations\Constraint\ConstraintMessageExtractor;
+use Doctrine\Common\Annotations\AnnotationReader;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassLike;
+use PhpParser\Node\Stmt\Property;
+
+/**
+ * Searches for constraints that are placed on a property.
+ */
+class PropertyValidationVisitor extends AbstractVisitor
+{
+    /**
+     * @var string[]
+     */
+    private $classStack = [];
+
+
+    /**
+     * @var AnnotationReader
+     */
+    private $reader;
+
+
+    /**
+     * @var ConstraintMessageExtractor
+     */
+    private $constraintMessageExtractor;
+
+
+    /**
+     *
+     */
+    public function __construct ()
+    {
+        $this->reader = new AnnotationReader();
+        $this->constraintMessageExtractor = new ConstraintMessageExtractor();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function enterNode (Node $node)
+    {
+        if ($node instanceof ClassLike)
+        {
+            if (!\property_exists($node, "namespacedName") || (!\is_string($node->namespacedName) && !$node->namespacedName instanceof Node\Name))
+            {
+                $this->classStack[] = "?";
+                return;
+            }
+
+            $this->classStack[] = (string) $node->namespacedName;
+            return;
+        }
+
+        if (!$node instanceof Property)
+        {
+            return;
+        }
+
+        if (count($node->props) !== 1)
+        {
+            throw new \Exception("Can't parse prop with multiple propertyproperties.");
```

Comment:
Am I correct in understanding that we're only capable of parsing

```php
class A
{
	/**
	 * @Asserts\NotBlank(message="...")
	 **/
	private $foo;
}
```

but not this?

```php
class B
{
	/**
	 * @Asserts\NotBlank(message="...")
	 * @Asserts\Length(max="1337", maxMessage="...")
	 **/
	private $bar;
}
```

Or is this specific line about nested properties, such as this?


```php
class C
{
     * @ORM\ManyToMany(targetEntity="...", inversedBy="...", cascade={"all"})
     * @ORM\JoinTable(
     *     name="...",
     *     joinColumns={
     *          @ORM\JoinColumn(name="..."),
     *     },
     *     inverseJoinColumns={
     *          @ORM\JoinColumn(name="..."),
     *     },
     * )
	private $baz;
}
```

Reply:
No, a property can actually wrap multiple properties:

```php
class A 
{
    public $a, $b, $c;
}
```

which is sort-of-the-same as

```php
class A
{
    public $a;
    public $b;
    public $c;
}
```

This is what `Property` and `PropertyProperty` represent.

This was explicitly disabled here, as the semantics are super confusing:


```
class A
{
    /**
     * @var
     * @Assert\NotBlank()
     */
    private $a, $b, $c;
}
```

in this case the `@NotBlank` annotation will **only belong to `$a` and not to the other properties**.
But as we are just interested in the error messages and not where they actually belong, we can allow this case. I just pushed the implementation for that.